pkg/autodiscovery/providers/prometheus_services.go

```diff
@@ -122,11 +228,43 @@ func (p *PrometheusServicesConfigProvider) invalidateIfChanged(old, obj interfac
 	// Compare annotations
 	if p.promAnnotationsDiffer(castedObj.GetAnnotations(), castedOld.GetAnnotations()) {
 		log.Trace("Invalidating configs on service change")
-		p.upToDate = false
+		p.setUpToDate(false)
 		return
 	}
 }
 
+func (p *PrometheusServicesConfigProvider) invalidateIfChangedEndpoints(old, obj interface{}) {
```

Comment:
this is practically the same method in `kube_endpoints.go`, can we refactor it by isolating the duplicated logic in a common helper?

Reply:
We should probably leave it for now - we know we need to refactor this code base. I could see a case for combining those two service providers into one.